R/common-find-variation.R

```diff
@@ -425,4 +425,498 @@ getPipedValue <- function(string) {
+                 rgb(255,127,85,maxColorValue = 255),
+                 rgb(245,176,236,maxColorValue = 255),
+                 rgb(108,77,23,maxColorValue = 255),
+                 rgb(255,170,196,maxColorValue = 255),
+                 rgb(210,199,132,maxColorValue = 255),
+                 rgb(157,25,89,maxColorValue = 255),
+                 rgb(171,105,96,maxColorValue = 255),
+                 rgb(122,62,101,maxColorValue = 255),
+                 rgb(255,133,152,maxColorValue = 255),
+                 rgb(135,60,42,maxColorValue = 255),
+                 rgb(137,56,65,maxColorValue = 255))
+    
+
+  if (printPlot == TRUE) {
+    for (i in 1:length(measureColumn)) {
+      model <- lm(df[[measureColumn[i]]] ~ interaction(l))
+      ANOVA <- aov(model)
+      # Tukey test to study each pair of treatment :
+      TUKEY <- TukeyHSD(x = ANOVA, conf.level = 0.95)
+      labs = generate_label_df(TUKEY, 'interaction(l)')
+      
+      par(bg = "transparent", cex.axis = 1)
+      a <- boxplot(df[[measureColumn[i]]]~interaction(l), data = df, col = my_colors[as.numeric(labs[,1])],
+                   ylab = measureColumn[i], ylim = c(min(df[measureColumn[[i]]], na.rm = TRUE), 
+                                                     1.1*max(df[[measureColumn[i]]], na.rm = TRUE)),
+                   cex.lab = 1.25)
+      # Now set the plot region to grey
+      rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "grey90")
+      grid(nx = NULL, ny = NULL, lwd = 1, lty = 3, col = "white") #grid over boxplot
+      par(new = TRUE)
+      a <- boxplot(df[[measureColumn[i]]]~interaction(l), data = df, col = my_colors[as.numeric(labs[,1])],
+                   ylab = measureColumn[i], ylim = c(min(df[measureColumn[[i]]]) , 1.1*max(df[[measureColumn[i]]])),
+                   main = paste("Boxplot of", measureColumn[i], "Across", paste(categoricalCols,collapse = " ")),
+                   cex.lab = 1.25, outcol = "lightcoral", add = TRUE)
+      over = 0.1*max(a$stats[nrow(a$stats),] )
+      text(c(1:nlevels(interaction(l))) , a$stats[nrow(a$stats),] + over, labs[,1],
+           col = my_colors[as.numeric(labs[,1])])
+      
+      
+      # plot 95% family-wise confidence level
+      if (length(TUKEY$'interaction(l)'[,4]) == 1) {
+        psig <- as.numeric(TUKEY$'interaction(l)'[,2]*TUKEY$'interaction(l)'[,3] >= 0 ) + 1
+      } else {
+        psig <- as.numeric(apply(TUKEY$'interaction(l)'[,2:3],1,prod) >= 0 ) + 1
+      }
+      
+      op <- par(mar = c(4.2,9,3.8,2))
+      plot(TUKEY,col = psig, yaxt = "n")
+      for (j in 1:length(psig)) {
+        axis(2,at = j,labels = rownames(TUKEY$'interaction(l)')[length(psig) - j + 1],
+             las = 1,cex.axis = .8,col.axis = psig[length(psig) - j + 1])
+      }
+      par(op)
+    } 
+  } else {
+    plotRes <- boxplot(df[[measureColumn[i]]]~interaction(l), data = df, plot = FALSE)
+  }
+  
+  # Return tables with mean/std and quartiles
```

Comment:
I think the return table should include the pvalue difference from the mean. And be sorted by significance.